src/hooks/useCreateQuiz.js

```diff
@@ -1,48 +1,93 @@
 import { useNavigate } from "react-router-dom";
 import axios from "../config/axios";
-import api from "../config/api";
+import api, { generateRoute } from "../config/api";
 import errorDisplayed from "../config/error";
 import useAuth from "./useAuth";
 /* eslint-disable no-unused-vars */
+/* eslint-disable prefer-template */
+/* eslint-disable no-underscore-dangle */
 
 const LOGIN_URL_EXPLORE = "/explore";
 
+// to do: clean urls
+
 export const useCreateQuiz = () => {
   const navigate = useNavigate();
   const { user } = useAuth();
 
   const createQuiz = async (values, onError) => {
     try {
-      const response = await axios.post(
+      const responseCreateQuiz = await axios.post(
         api.quizzes,
         JSON.stringify({ id_user_owner: user.userId,
           title: values.title,
           description: values.description,
-          categories: values.categories,
-          /* questions: values.questions */ }),
+          categories: values.category }),
+        {
+          headers: { Authorization: `Bearer ${user.token}`,
+            "Content-Type": "application/json" },
+          withCredentials: true
+        }
+      );
+
+      console.log("responseCreateQuizSkeleton", responseCreateQuiz);
+
+      const convertedQuestions = [];
+      values.questions.map((question, position) => convertedQuestions.push({
+        id_quiz: responseCreateQuiz.data.idQuiz,
+        title: question.question,
+        choices: [],
+        explanation: question.explanation,
+        link_to_learn_more: question.learnMore,
+        position
+      }));
+
+      console.log("convertedQuestions", convertedQuestions);
+
+      const responseCreateQuestions = await axios.post(
+        generateRoute(api.questions, responseCreateQuiz.data.idQuiz),
+        JSON.stringify({ questions: convertedQuestions }),
         {
           headers: { Authorization: `Bearer ${user.token}`,
             "Content-Type": "application/json" },
           withCredentials: true
         }
       );
+      console.log("responseCreateQuestions", responseCreateQuestions);
+      console.log("responseCreateQuestions.data.idQuestions", responseCreateQuestions.data.idQuestions);
 
-      /* console.log(response.data);
-      console.log(JSON.stringify(response)); */
+      responseCreateQuestions.data.idQuestions.map(async (idQuestion, i) => {
```

Comment:
why do we need async here? :)

Reply:
because this map is allowing us to asynchronously add Choice in the database. You can see a bit under that we are indeed using an "await" to post each Choice with axios